File: CSharpPractice/LeetcodePractice/Leetcode134.cs
public class Solution134 {
    public int CanCompleteCircuit(int[] gas, int[] cost) 
    {

        if(gas.Sum()-cost.Sum()<0){return -1;}
        int ans=0;
        int curGas=0;
        for(int i=0;i<gas.Length;i++)
        {
            curGas +=(gas[i]-cost[i]);
            if(curGas<0)
            {
                ans=i+1;
                curGas=0;
            }

        }
        return ans;
        
    }
}

File: JavaScriptPractice/Leetcode134.js
var canCompleteCircuit = function(gas, cost) 
{
    if(gas.reduce((a,b)=>a+b,0)-cost.reduce((a,b)=>a+b,0)<0){return -1;}
    let ans = 0;
    let curGas = 0;
    for(let i=0;i<cost.length;i++)
    {
        curGas +=(gas[i]-cost[i]);
        if(curGas<0)
        {
            curGas=0;
            ans=i+1;
        }

    }
    return ans;
};